test/controllers/sessions_controller_test.rb

```diff
@@ -0,0 +1,36 @@
+require "test_helper"
+
+describe SessionsController do
+  it "gets login form" do
+    get login_path
+    must_respond_with :success
+  end
+
+  it "login for existing user" do
+    proc {
+      post login_path, params: {name: users(:averi).name}
+    }.must_change 'User.count', 0
+    session[:user].must_equal users(:averi).name
+    session[:user_id].must_equal users(:averi).id
+  end
+
+  it "login for new user" do
+    proc {
+      post login_path, params: {name: "NEW NAME"}
+    }.must_change 'User.count', 1
+  end
+
+  it "won't login for invalid user" do
+    proc {
+      post login_path, params: {name: nil}
+    }.must_change 'User.count', 0
+  end
+
+  it "will log out a user" do
+    delete login_path
+    assert_nil session[:user]
```

Comment:
This works fine, but doing it spec-style would look like:

`session[:user].must_be_nil`